src/core/config/descriptor_providers/xlio_config_schema.json

```diff
@@ -0,0 +1,1454 @@
+                                    "maximum": 65280,
+                                    "title": "RX buffer size",
+                                    "description": "Size of Rx data buffer elements allocation. Cannot be less than MTU (Maximum Transfer Unit) and greater than 0xFF00. Default value is calculated based on maximum MTU. Maps to XLIO_RX_BUF_SIZE environment variable."
+                                },
+                                "override_rcvbuf_limit": {
+                                    "type": "integer",
+                                    "default": 65536,
+                                    "title": "Override OS receive buffer limit",
+                                    "description": "Minimum value in bytes that will be used per socket by XLIO when applications call to setsockopt(SO_RCVBUF). If application tries to set a smaller value than configured here, XLIO will force this minimum limit value on the socket. XLIO offloaded socket's receive max limit of ready bytes count. If the application does not drain a socket and the byte limit is reached, new received datagrams will be dropped. Monitor of the applications socket's usage of current, max and dropped bytes and packet counters can be done with xlio_stats. Maps to XLIO_RX_BYTES_MIN environment variable."
+                                },
+                                "prefetch_size": {
+                                    "type": "integer",
+                                    "default": 256,
+                                    "minimum": 32,
+                                    "title": "RX prefetch size",
+                                    "description": "Size of receive buffer in bytes to prefetch into cache while processing ingress packets. Maps to XLIO_RX_PREFETCH_BYTES environment variable. The default is a single cache line of 64 bytes which should be at least 32 bytes to cover the IP+UDP headers and a small part of the users payload. Increasing this can help improve performance for larger user payload sizes.\nValue range is 32 bytes to MTU size"
+                                },
+                                "prefetch_before_poll": {
+                                    "type": "integer",
+                                    "default": 0,
+                                    "title": "Prefetch before polling",
+                                    "description": "Same as RX prefetch size, only that prefetch is done before actually getting the packets. This benefits low pps traffic latency. Disable with 0. Maps to XLIO_RX_PREFETCH_BYTES_BEFORE_POLL environment variable."
+                                }
+                            }
+                        },
+                        "work_request_elements": {
+                            "type": "object",
+                            "description": "Work Request Element settings.",
+                            "properties": {
+                                "tx_global_array_size": {
```

Comment:
This one is not global. its per SQ. maybe all this section should be under ring? anyway its tx_ring_elements_count